src/schdispatch_cmd.cpp

```diff
@@ -376,10 +377,19 @@ CommandCost CmdScheduledDispatchAddNewSchedule(TileIndex tile, DoCommandFlag fla
 	if (flags & DC_EXEC) {
 		v->orders->GetScheduledDispatchScheduleSet().emplace_back();
 		DispatchSchedule &ds = v->orders->GetScheduledDispatchScheduleSet().back();
-		ds.SetScheduledDispatchDuration(p2);
-		ds.SetScheduledDispatchStartTick((StateTicks)p3);
-		ds.UpdateScheduledDispatch(nullptr);
-		SetTimetableWindowsDirty(v, STWDF_SCHEDULED_DISPATCH);
+		if (scheduleJson == nullptr) {
+			if (p2 == 0) return CMD_ERROR;
```

Comment:
This check shouldn't be within DC_EXEC because text and exec runs should return the same command result (except where flagged otherwise).